stack/stack.go

```diff
@@ -200,22 +200,28 @@ func ErrStack() error {
 	return callers(3)
 }
 
+// Errorf formats an error using fmt.Errorf and joins it with
+// a stack if the error does not already have one.
+func Errorf(format string, a ...any) error {
+	return joinErrors(nil, fmt.Errorf(format, a...))
+}
```

Comment:
Can we get this put into the main `errdefs` package as something like `errdefs.New`? Or something like `errdefs.Errorf`. I think this is the functionality we want, but I'd like it if it were easier to access and didn't require using a subpackage especially since `stack` is a bit of a generic name.

Reply:
ping @dmcgowan